enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,489 @@
+capability must also be applied to the cluster, regardless of the expressed configuration.
+7) Capabilities can be enabled post-install, but they cannot be disabled post-install (it is too difficult
+to safely/fully clean up the cluster)
+
+
+## Motivation
+
+There is an increasing desire to move away from "one size fits all" cluster installations, and
+towards flexibility about what should/should not exist in a new cluster out of the box.  This can
+be seen in efforts such as hypershift, single node, and code-ready-containers.  Each of these
+efforts has done some amount of one-off work to enable their requirements.  This EP proposes a
+mechanism that allows components to be disabled in a first class way that the installer exposes.
+
+### Goals
+
+* Admins can easily explicitly exclude specific "optional" components from their cluster, at install time.
+* Admins can enable a previously excluded optional component, at runtime.
+* Install wrappers like assisted-installer can define an install-config that excludes specific components
+* Define an api that could be used in the future to exclude cluster capabilities based on things other than
+CVO filtering, such as turning off a particular api for the cluster.
+
+### Non-Goals
+
+* Making control-plane critical components optional (k8s apiserver, openshift apiserver, openshift controller,
+  networking, etc)
+* Defining which components should be disable-able (this will be up to component teams to classify themselves
+as `capabilities` or not)
+* Providing a way to install OLM operators as part of the initial cluster install.  This EP is about making
+the install experience around the existing CVO-based components more flexible, not adding new components to the
+install experience.
+* Allowing components to be disabled post-install.
+* Eliminating or replacing cluster profiles
+* Encoding logic in the installer itself about which components should be disabled under specific circumstances
+
+
+## Proposal
+
+### User Stories
+
+* As a user creating a new cluster that will be managed programmatically, I do not want the additional
+security exposure and resource overhead of running the web console.  I would like a way to install
+a cluster that has no console out of the box, rather than having to disable it post-install or
+modify rendered manifests in a way that requires deep understanding of the OCP components/resources.
+
+* As a team scaffolding a managed service based on openshift, I want to minimize the footprint of my
+clusters to the components I need for the service.
+
+* As a user creating a cluster that will never run an image registry, I do not want the additional overhead
+of running the image registry operator, or have to remove the default registry that is created.
+
+* As a team using openshift for a specific use case such as edge deployments, I want to provide
+an install experience that disables components that aren't needed for my use case.
```

Comment:
i guess i'd rather not, mostly because i haven't thought deeply about what it would take to truly disable everything that OKE currently leaves enabled but tells you not to use.  I worry that it sends us down an unrelated rabbit-hole of problems that this EP isn't intended to cover.

(e.g. as you note, it would require making the openshift-apiserver optional.  At which point we need to figure out what happens when a customer disables the openshift-apiserver, but doesn't disable all the components that try to create Route + Imagestream resources)

could this eventually evolve to answer that?  yes.